src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1538,6 +1529,79 @@ bool RegisterAppInterfaceRequest::GetDataOnSessionKey(
   return true;
 }
 
+mobile_apis::Result::eType
+RegisterAppInterfaceRequest::ProcessingAppHMITypesInMessage(
+    const smart_objects::SmartObject& message) {
+  const bool app_hmi_types_exist =
+      message[strings::msg_params].keyExists(strings::app_hmi_type);
+
+  const auto result = mobile_apis::Result::SUCCESS;
+
+  if (!app_hmi_types_exist) {
+    return result;
+  }
+
+  auto app_types_in_message =
+      *(message[strings::msg_params][strings::app_hmi_type].asArray());
+
+  auto it = std::find(app_types_in_message.begin(),
+                      app_types_in_message.end(),
+                      mobile_apis::AppHMIType::WEB_VIEW);
+  if (app_types_in_message.end() != it) {
+    return mobile_apis::Result::DISALLOWED;
+  }
+
+  return result;
+}
+
+mobile_apis::Result::eType
+RegisterAppInterfaceRequest::ProcessingAppHMITypesPolicies(
```

Comment:
I think the added code for checking the app hmi type is redundant to existing means core has for validating app hmi type via policies.

Reply:
Instead of calling ProcessingAppHMITypesInMessage and ProcessingAppHMITypesPolicies can PolicyHandler::CheckHMIType be used instead?

Reply:
@JakLivio  `PolicyHandler::CheckHMIType` has a different intention. It used for processing RPC's if application is already registered. 
PolicyHandler::CheckHMITyp uses **either** policy_hmi_types **or** application hmi_types.
Please see simplified version of `PolicyHandler::CheckHMIType` : 
```cpp
bool PolicyHandler::CheckHMIType(application_id,  type_to_check, app_types) {
  policy_hmi_types =  policy_manager_->GetHMITypes(application_id);
  if (app_types) {
    return std::find(app_types.begin(), app_types.end(), type_to_check) != app_types.end();
  }
  return std::find(policy_hmi_types.begin(), policy_hmi_types.end(), type_to_check) != policy_hmi_types.end();
}
```

But  RAI::ProcessingAppHMITypesPolicies is checking that each HMI type from RAI is presented in policies, and combines response info for this. 
We can reuse `PolicyHandler::CheckHMIType` to avoid some code duplication, but I am not sure, that it will be much better.  Let me check

Reply:
@JackLivio  please check the commit with usage of PolicyHandler::CheckHMIType 2c695099a where it could be done in RAI

Reply:
Thanks for making the changes but I believe you are correct. Its not much cleaner and performance will be worse.

Reply:
I will revert the changes.